### PR TITLE
[SPARK-23094][SPARK-23723][SPARK-23724][SQL][FOLLOW-UP] Support custom encoding for json files

### PR DESCRIPTION
**Pull request by gatorsmile**
## What changes were proposed in this pull request?
This is to add a test case to check the behaviors when users write json in the specified UTF-16/UTF-32 encoding with multiline off.

## How was this patch tested?
N/A

### CODE REVIEW

**Comment by gatorsmile** [comment]
cc @MaxGekk @HyukjinKwon Do we have any behavior change after the previous PR: https://github.com/apache/spark/pull/20937?

**Comment by HyukjinKwon** [comment]
Nope, I am quite sure that we don't have any kind of hidden behaviour change. Both `lineSep` and `encoding` options are new. Also, these are actually now restricter than it's actually needed for now. Writing can actually work and https://github.com/apache/spark/pull/21247 tries to allow it; however, I left a comment for him to just focus on getting rid of the restrictions, which is his final goal in 2.4.0 (or 3.0.0).

**Review by HyukjinKwon — APPROVED** [review]
adding the test seems ok if you feel in that way. this might have to be removed soon within the next release since we should allow this case anyway.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala [hunk 1]
```diff
@@ -2313,6 +2313,25 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
     }
   }
 
+  test("SPARK-23723: write json in UTF-16/32 with multiline off") {
+    Seq("UTF-16", "UTF-32").foreach { encoding =>
+      withTempPath { path =>
+        val ds = spark.createDataset(Seq(
+          ("a", 1), ("b", 2), ("c", 3))
+        ).repartition(2)
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala]: we don't have to repartition though.

**Comment by MaxGekk** [comment]
> Do we have any behavior change after the previous PR: #20937?

The PR brought the `encoding` (and `charset`) option but we didn't change behavior when `encoding` is not specified.

As @HyukjinKwon wrote above the PR #21247 eliminates restrictions in write but the restrictions don't break previous behavior (before #20937) in any case.

**Comment by HyukjinKwon** [comment]
retest this please

**Pull request closed without merging**